bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -43,109 +45,201 @@ class RequestValidationService(
 
-        val legalAddressBridges = taskReservationRequests
-            .map {
-                AddressBridge(
-                    address = it.businessPartner.legalEntity?.legalAddress!!,
-                    request = it,
-                    bpnA = it.businessPartner.legalEntity?.legalAddress?.bpnAReference?.referenceValue!!
-                )
-            }
+        val requestsWithMainAddressNull = taskEntries.filter {
+            it.businessPartner.site != null && it.businessPartner.site?.mainAddress == null
+        }
+        val mainAddressEmptyErrorsByRequest = requestsWithMainAddressNull.associateWith {
+            listOf(ErrorInfo(SiteCreateError.MainAddressIdentifierNotFound, "Site main address is Empty", it.getRequestKey()))
+        }
 
-        val additionalAddressBridges = taskReservationRequests.filter { it.businessPartner.site == null && it.businessPartner.address != null }
-            .map {
-                AddressBridge(
-                    address = it.businessPartner.address!!,
-                    request = it,
-                    bpnA = it.businessPartner.address?.bpnAReference?.referenceValue,
-                )
+        val sitesToUpdate = taskEntries
+            .filter {
+                isSiteUpdate(it.businessPartner, requestMappings)
             }
 
-        val leErrorsByRequest = validateLegalEntitiesToUpdate(taskReservationRequests.map {
-            LegalEntityUpdateBridge(
-                legalEntity = it.businessPartner.legalEntity!!,
-                request = it,
-                bpnL = it.businessPartner.legalEntity?.bpnLReference?.referenceValue!!
-            )
+        val siteErrorsByRequest = validateSitesToUpdate(sitesToUpdate.map {
+            SiteUpdateBridge( request = it, bpnS = requestMappings.getBpn(it.businessPartner.site?.bpnSReference!!)!! )
         })
-        val addressErrorsByRequest = validateAddresses(legalAddressBridges + additionalAddressBridges, leUpdateMessages)
-        return mergeMapsWithCollectionInValue(leErrorsByRequest, addressErrorsByRequest)
+
+        return mergeMapsWithCollectionInValue(siteErrorsByRequest, mainAddressEmptyErrorsByRequest)
     }
 
-    fun validateSitesToCreateFromOrchestrator(taskReservationRequests: List<TaskStepReservationEntryDto>): Map<RequestWithKey, Collection<ErrorInfo<SiteCreateError>>> {
+    private fun validateAddressesFromOrchestrator(taskEntries: List<TaskStepReservationEntryDto>,  requestMappings: TaskEntryBpnMapping): Map<RequestWithKey, Collection<ErrorInfo<ErrorCode>>> {
 
-        // parents have to be checked in the legal entity validation
-        val (requestsWithMainAddressNull, requestsWithMainAddressNotNull) = taskReservationRequests.partition { it.businessPartner.site?.mainAddress == null }
+        val taskWithLegalAddress = taskEntries
```

Comment:
Why this filter condition? 
Is it only about legal addresses, as the variable name suggests?
What about pure additional addresses, not legal or main addresses?
I find this address validation method and its sub-methods confusing.

Reply:
This filter is a precondition for the following methods. Only addresses that have a LegalAddress and that have a MainAdress if a site is present may be considered.
As mentioned in the description , the check for duplicates has to be done over all inserted/updated addresses. So we have to collect first alle addresses.